src/engine/platform/mixer.cpp

```diff
@@ -47,6 +55,53 @@ struct AudioStruct {
     SDL_AudioDeviceID dev;
 } audio_struct = {};
 
+Effect create_delay(f32 feedback, f32 delay_time) {
+    ASSERT(delay_time < CHANNEL_BUFFER_LENGTH_SECONDS, "Delay time is longer than channel buffer");
+    auto delay_func = [] (Effect *effect, f32 *buffer, u32 start, u32 len) -> void {
+        if (effect->delay._delay_len_seconds != effect->delay._prev_delay_len_seconds) {
+            //TODO(GS) crackling when length is changed while sound is playing
```

Comment:
Solving this would be challengeing, I would recomend having a bool that says something like "phasing out", which gradually turns down the effect strength.